app/routes.php

```diff
@@ -83,6 +83,10 @@
             ->add(PersonWithPasswordAuthMiddleware::class)
             ->add($ipMiddleware)
             ->add(RateLimitMiddleware::class);
+        $versionGroup->get('/regular-giving/my-donation-mandates/{mandateId:[a-z0-9-]{36}}/cancel', RegularGivingMandate\Cancel::class)
+            ->add(PersonWithPasswordAuthMiddleware::class)
+            ->add($ipMiddleware)
+            ->add(RateLimitMiddleware::class);
```

Comment:
may be worth refactoring at this point (or when this is being manually tested with FE implementation) to combine with other routes into a group with shared middlewares.

Also (and I'm sure I already wrote this comment but now can't find it) not sure if it's really worthwhile having the mandateId in both the URI and the request body. It allows checking that they match in the controller but not sure that really does much, both are likely to be set based on the same variable in FE. What do you think @NoelLH ?

Reply:
I do find it helpful to have IDs in the route for things like this @bdsl; it quite frequently makes it easier to work with Apache logs (as well as following what I've historically understood to be RESTful API design received wisdom). But it's also a bit odd and annoying to work with an object that's missing its most fundamental property for unique identification. So although I don't like the redundancy of doing both, I think it might still be the least bad option.

Reply:
OK, will leave it having both.